File: Panta-Pincay-Leccion/src/app/interfaces/iproducto.ts
export interface TipoProducto {
    id:         number
    nombre:     string
}

export interface IProducto {
    id?:        number
    codigo:     string
    nombre:     string
    tipo:       TipoProducto
    precio:     number
    cantidad:   number
    subtotal:   number
}

File: SullyPanta-Leccion/SullyPanta-Leccion/Models/Iproducto.cs
using System;
using System.Collections.Generic;

namespace SullyPanta_Leccion.Models;

public partial class Iproducto
{
    public int Id { get; set; }

    public string Codigo { get; set; } = null!;

    public string Nombre { get; set; } = null!;

    public int Tipo { get; set; }

    public decimal Precio { get; set; }

    public int Cantidad { get; set; }

    public decimal Subtotal { get; set; }

    public virtual TipoProducto TipoNavigation { get; set; } = null!;
}
